src/fixit/rules/deprecated_abc_import.py

```diff
@@ -0,0 +1,250 @@
+                    and statement.module
+                    and statement.module.value == "collections"
+                    # Ensure the imports in this statement are the ones we are looking for
+                    and set(import_names).intersection(set(self.imports_names))
+                ):
+                    return statement
+
+        return None
+
+    def leave_Module(self, node: cst.Module) -> None:
+        """
+        While leaving the module, check if we need to split up imports.
+        """
+        if self.update_module:
+            # Filter the ABCs and non-ABCs
+            abcs: List[str] = []
+            non_abcs: List[str] = []
+            for name in self.imports_names:
+                (non_abcs, abcs)[name in ABCS].append(name)
+
+            node_body = list(node.body)
+
+            # Iterate over the module to find bad imports
+            for idx, statement in enumerate(node_body):
+                # Find if the statement is the one we are searching for
+                import_statement = self.get_import_from(statement)
+                if import_statement:
+                    # Remove the original import statement
+                    node_body.remove(statement)
+                    # Add the non ABC imports
+                    node_body.insert(
+                        idx,
+                        cst.SimpleStatementLine(
+                            body=(
+                                cst.ImportFrom(
+                                    module=cst.Name(value="collections"),
+                                    names=[
+                                        cst.ImportAlias(name=cst.Name(value=imp))
+                                        for imp in non_abcs
+                                    ],
+                                ),
+                            )
+                        ),
+                    )
+                    # Add the ABC imports
+                    node_body.insert(
+                        idx + 1,
+                        cst.SimpleStatementLine(
+                            body=(
+                                cst.ImportFrom(
+                                    module=cst.Attribute(
+                                        value=cst.Name(value="collections"),
+                                        attr=cst.Name(value="abc"),
+                                    ),
+                                    names=[
+                                        cst.ImportAlias(name=cst.Name(value=imp))
+                                        for imp in abcs
+                                    ],
+                                ),
+                            )
+                        ),
+                    )
+
+            self.report(node, replacement=node.with_changes(body=node_body))
+
+    def visit_ImportAlias(self, node: cst.ImportAlias) -> None:
+        """
+        This catches the `import collections.<ABC>` cases.
+        """
+        if (
```

Comment:
Not sure if it would be more clear to express this big conditional using a matcher decorator: https://libcst.readthedocs.io/en/latest/matchers_tutorial.html#Matcher-Decorators

Reply:
Updated to use matcher in the conditional. Couldn't get decorators to function properly,